### ISSUE TITLE
Trigger for Open Merge Requests non-functional in pipeline jobs

### ISSUE THREAD

**Issue by czoellin** [jenkinsci/gitlab-plugin#515]
### Context
- **Gitlab plugin version**: 1.4.6
- **Gitlab version**: 8.16.1
- **Jenkins version**: 2.45
- **Job type**: Pipeline

### Problem Description

There is a "bug" in the OpenMergeRequestPushHookTriggerHandler. The handler assumes that the Job is an instance of the AbstractProject class. 
```java
            if (job instanceof AbstractProject<?, ?>) {
                AbstractProject<?, ?> project = (AbstractProject<?, ?>) job;
```

However, apparently (I'm not an expert in Jenkins) pipeline/workflow jobs do not inherit from that class. Instead workflow jobs re-implement the interface related to triggers (ParameterizedJobMixIn.ParameterizedJob).

Since there's no reason to rely on AbstractProject, I refactored this to use the common interface method:
```java
            if (job instanceof ParameterizedJobMixIn.ParameterizedJob) {
                ParameterizedJob project = (ParameterizedJobMixIn.ParameterizedJob) job;
                GitLabConnectionProperty property = job.getProperty(GitLabConnectionProperty.class);
                for (Trigger t : project.getTriggers().values()) {
                	if (t instanceof GitLabPushTrigger) {
                		final GitLabPushTrigger trigger = (GitLabPushTrigger) t;
```

I'm going to supply a merge request with the fix.

**Comment by jgeorgeson** [comment]
(not a project member) What versions are you using? I am running with these versions

Jenkins: 2.47
GitLab CE: 8.16.5
GitLab Jenkins Plugin: 1.4.5

In my environment I found that with a single Job that is configured to trigger on both Push and Merge Request events, and a GitLab Web Hook configured to send on both events, a push to the source branch of an open MR resulted in two builds being run.

**Comment by czoellin** [comment]
@jgeorgeson seems unrelated to this issue. Is this a bug report for my PR?

Don't have access to the exact version numbers right now. But only the gitlab plugin is the very latest version.

**Comment by omehegan** [comment]
This was released today in version 1.4.6 of the plugin.
